src/contexts/tokens.tsx

```diff
@@ -57,26 +57,41 @@ export function TokensProvider({ children = undefined }) {
   }
 
   const filteringChainId = useMemo(() => {
-    if (Number.isNaN(chainId)) return CHAIN_IDS['devnet']
-    return chainId
-  }, [chainId])
+    if (currentProvider) {
+      if (currentProvider && Number.isNaN(currentProvider.networkVersion)) {
```

Comment:
@Tkachez, could you check this condition? For wallets like Rabby or Trust it wouldn't work. Change this `currentProvider.networkVersion`.